### PR TITLE
Include Repository Schemas from Adobe Cloud Platform + HAL

### PR DESCRIPTION
**Pull request by trieloff**
No longer a work in progress. The PR will model the directory and asset representations of ACP, and use them as base classes for XDM asset and XDM content.

This PR subsumes PR #70.

Quick summary of changes:

- introduces HAL for expressing links in `schemas/external/hal`
- introduces `schemas/external/repo` for ACP storage specific schemas
- models ACP storage, i.e. `asset`, `directory`, and `sub-directory` according to ACP spec
- changes existing `content` and `asset` to inherit from ACP `asset`
- apply trickle-down changes to everything that inherits from `asset`, i.e. `image` and `video`
- apply trickle-down changes to all examples

@ksteeter please have a look.

### CODE REVIEW

File: schemas/external/repo/directory.schema.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  ],
+  "definitions": {
+    "directory": {
+      "properties": {
+        "dc:format": {
+          "type": "string",
+          "title": "Format",
+          "const": "application/vnd.adobecloud.directory+json"
+        },
+        "children": {
```
Comment (lucianfelix) [inline schemas/external/repo/directory.schema.json]: Shouldn't this belong to HAL _links? Or to _embedded if simple children representation needs to be included?

File: schemas/external/repo/asset.example.1.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (lucianfelix) [inline schemas/external/repo/asset.example.1.json]: Would be useful to include HAL properties in some examples.

File: schemas/external/repo/directory.schema.json (hunk 1 above)
Comment (trieloff) [reply schemas/external/repo/directory.schema.json]: Not according to [ACP spec](https://git.corp.adobe.com/pages/AdobeCloudPlatform/api-spec/repository_api/hypermedia.html), which I just tried to model here.

File: schemas/external/repo/asset.example.1.json (hunk 2 above)
Comment (trieloff) [reply schemas/external/repo/asset.example.1.json]: ✅ f39bc9e

**Pull request merged**